sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/MethodsShouldUseBaseTypes.cs

```diff
@@ -342,9 +346,12 @@ ITypeSymbol FindMostGeneralAccessibleClassOrSelf(ITypeSymbol typeSymbol)
                         if (DerivesOrImplementsAll(currentSymbol))
                         {
                             typeSymbol = currentSymbol;
+                            currentSymbol = currentSymbol.BaseType;
```

Comment:
This is just a performance optimization to end the loop early. There's no need to re-check base types with `DerivesOrImplementsAll` once we're sure that the type itself doesn't conform.